File: CovidWatch/app/src/main/java/com/riskre/covidwatch/data/CovidWatchRepository.java
package com.riskre.covidwatch.data;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.riskre.covidwatch.CovidWatchApplication;

import java.util.List;

class CovidWatchRepository {

    private ContactEventDAO cenDAO;
    private LiveData<List<ContactEvent>> allEvents;

    CovidWatchRepository(Application application) {
        CovidWatchDatabase db = CovidWatchDatabase.getDatabase(application);
        cenDAO = db.contactEventDAO();
        allEvents = cenDAO.getAllSortedByDescTimestamp();
    }

    /**
     * Room executes all queries on a separate thread.
     * Observed LiveData will notify the observer when the data has changed.
     */
    LiveData<List<ContactEvent>> getAllEvents() {
        return allEvents;
    }

    /**
     * You must call this on a non-UI thread or your app will throw an exception. Room ensures
     * that you're not doing any long running operations on the main thread, blocking the UI.
     */
    void insert(ContactEvent cen) {
        CovidWatchDatabase.databaseWriteExecutor.execute(() -> {
            cenDAO.insert(cen);
        });
    }
}

File: CovidWatch/app/src/main/java/org/covidwatch/android/data/CovidWatchRepository.kt
package org.covidwatch.android.data

import android.app.Application
import androidx.lifecycle.LiveData


internal class CovidWatchRepository(application: Application) {

    private val contactEventDAO: ContactEventDAO

    /**
     * TODO
     * Room executes all queries on a separate thread.
     * Observed LiveData will notify the observer when the data has changed.
     *
     * @return allEvents, all the contact events in the DB
     */
    val allEvents: LiveData<List<ContactEvent>>

    /**
     * You must call this on a non-UI thread or your app will throw an exception. Room ensures
     * that you're not doing any long running operations on the main thread, blocking the UI.
     */
    fun insert(cen: ContactEvent) {
        CovidWatchDatabase.databaseWriteExecutor.execute { contactEventDAO.insert(cen) }
    }

    init {
        val db = CovidWatchDatabase.getInstance(application)
        contactEventDAO = db.contactEventDAO()
        allEvents = contactEventDAO.allSortedByDescTimestamp
    }
}